src/enclosure.c

```diff
@@ -171,3 +184,154 @@ int enclosure_open(const struct enclosure_device *enclosure)
 
 	return fd;
 }
+
+static struct enclosure_device *find_enclosure(char *enclosure_id)
+{
+	struct enclosure_device *enclosure_device;
+	list_for_each(sysfs_get_enclosure_devices(), enclosure_device) {
+		if (strncmp(enclosure_device->dev_path, enclosure_id, PATH_MAX) == 0) {
+			return enclosure_device;
+		}
+	}
+	return NULL;
+}
+
+status_t find_block_device_by_enclosure_slot(struct enclosure_device *encl, int index, enum ibpi_pattern *status, struct block_device **device)
+{
+	*status = IBPI_PATTERN_NORMAL;
+	*device = NULL;
+	for (int i = 0; i < encl->slots_count; i++) {
+		if (encl->slots[i].index == index) {
+			*status = encl->slots[i].ibpi_status;
+			*device = locate_block_by_sas_addr(encl->slots[i].sas_addr);
+			return STATUS_SUCCESS;
+		}
+	}
+	return STATUS_DATA_ERROR;
+}
+
+status_t enclosure_get_slot_by_device(char *device, struct slot_response *slot_res)
```

Comment:
If we have a `device` we probably can obtain a slot. I would prefer to obtain a slot and call `enclosure_get_slot_by_slot_num` later.
Now, we have to separate logics for device and slot, generally it brings possibility to have two different outputs.
For example, for NPEM we have only one function:
`status_t npem_get_slot(char *device, char *slot_path, struct slot_response *slot_res)`

Is it possible in this case?

Reply:
See updated implementation